File: src/main/java/io/github/tanguygab/conditionalactions/actions/items/RepairItemAction.java
package io.github.tanguygab.conditionalactions.actions.items;

import io.github.tanguygab.conditionalactions.actions.Action;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;
import org.bukkit.inventory.meta.Damageable;

public class RepairItemAction extends Action {

    public RepairItemAction() {
        super("(i?)repair-item:( )?");
    }

    @Override
    public String getSuggestion() {
        return "repair-item: <slot>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        if (!(player instanceof Player p)) return;

        int slot = parseInt(match,-1);
        PlayerInventory inv = p.getInventory();
        if (slot == -1) slot = inv.getHeldItemSlot();

        ItemStack item = inv.getItem(slot);
        if (item == null || item.getType().isAir()) return;
        if (item.getItemMeta() instanceof Damageable d && d.hasDamage()) {
            d.setDamage(0);
            item.setItemMeta(d);
        }
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/items/RepairItemAction.kt
package io.github.tanguygab.conditionalactions.actions.types.items

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player
import org.bukkit.inventory.meta.Damageable

class RepairItemAction : Action("^(i?)repair-item:( )?") {
    override val suggestion = "repair-item: <slot>"

    override fun execute(player: OfflinePlayer?, match: String) {
        if (player !is Player) return

        var slot = parseInt(match, -1)
        val inv = player.inventory
        if (slot == -1) slot = inv.heldItemSlot

        val item = inv.getItem(slot)
        if (item == null || item.type.isAir) return
        val meta = item.itemMeta
        if (meta is Damageable && meta.hasDamage()) {
            meta.damage = 0
            item.itemMeta = meta
        }
    }
}
